### PR TITLE
Pitching Data

### PR DESCRIPTION
**Pull request by MitchThompson2001**
Created HTML page and logic for pitching table. Also updated import statements to be complete

### CODE REVIEW

File: Team_Roster/app/routes.py [hunk 1]
```diff
@@ -16,6 +18,11 @@ def roster(teamId):
     year =request.args.get("year")
     return ShowRoster(teamId,year)
 
+@app.route('/<teamId>/depthChart', methods=['GET'])
```
Comment (Jarpletz) [inline Team_Roster/app/routes.py]: why did you add a route here? The pitching data can just go in the roster page, using the following files:
```
roster.py
roster.html
```

File: Team_Roster/app/templates/pitchingChart.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{% extends "teamContainer.html" %}
+
+{% block content %}
+<h1>Pitching Statistics for Team {{ teamId }} ({{ year }})</h1>
+
+<table>
```
Comment (Jarpletz) [inline Team_Roster/app/templates/pitchingChart.html]: Table looks good, but please move it to roster.py.

File: Team_Roster/pages/pitchingStats.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+from flask import render_template
+from sqlalchemy import func
+from Team_Roster.app import db
+from Team_Roster.app.models import Pitching, People
+
+def ShowPitchingChart(teamId, year):
```
Comment (Jarpletz) [inline Team_Roster/pages/pitchingStats.py]: Move to roster.py, maybe create another function in that file that returns pitching_data and is called in `showRoster`.

File: Team_Roster/pages/pitchingStats.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            func.div(func.sum(Pitching.p_BB) * 100.0, func.sum(Pitching.p_BFP)).label("BB%"),
+            func.div(func.sum(Pitching.p_HR) * 9.0,
+                     func.div(func.sum(Pitching.p_IPOuts), 3.0)).label("HR/9"),
+            func.div(
+                (func.sum(Pitching.p_H) - func.sum(Pitching.p_HR)),
+                (func.sum(Pitching.p_BFP) - func.sum(Pitching.p_SO) -
+                 func.sum(Pitching.p_BB) - func.sum(Pitching.p_HR) - func.sum(Pitching.p_HBP))
+            ).label("BABIP"),
+            func.div(
+                (func.sum(Pitching.p_H) + func.sum(Pitching.p_BB) +
```
Comment (Jarpletz) [inline Team_Roster/pages/pitchingStats.py]: great job on all the calculations!!!!

File: Team_Roster/app/__init__.py [hunk 5]
```diff
@@ -9,4 +9,4 @@
 migrate = Migrate(app, db)
 
 
-from app import routes
\ No newline at end of file
+from Team_Roster.app import routes
```
Comment (Jarpletz) [inline Team_Roster/app/__init__.py]: Just curious, what is your reason for making this change?

**Review by Jarpletz — CHANGES_REQUESTED** [review]
Overall looks good! We can add all of this data to the roster page (`roster.py` and `roster.html` instead of having a separate batting data page). Otherwise, looks great to me, good job following existing patterns and doing all those confusing stats calculations! Thank you for your hard work!

File: Team_Roster/app/__init__.py (hunk 5 above)
Comment (MitchThompson2001) [reply Team_Roster/app/__init__.py]: My code was having an issue finding the path to app, but specifying the directory helped

**Review by Jarpletz — CHANGES_REQUESTED** [review]
Overall this looks pretty good! However, I think there might be an error in how FIP and WAR are being calculated - looking at fanGraphs, its showing values between -1 and 10 for WAR, and between 3 and 10 for FIP. However, it looks like our tables are ranging from at least -7000 to 350 for both of these stats, which probably isn't right. Could you please investigate this? Otherwise looks good, great work!

**Comment by Jarpletz** [comment]
>... and between 3 and 10 for FIP. However, it looks like our tables are ranging from at least -7000 to 350 for both of these stats, which probably isn't right.

I have pushed into main, in the depth chart tables, a correct calculation for (Pitching) FIP. You may want to reference off of this, the equation is similar, if not the same, and requries an "FIP Constant" that I added via a new table in the database.
